### ISSUE TITLE
EfficientNet extremely slow

### ISSUE THREAD

**Issue by tmabraham** [pytorch/xla#2715]
I have recently been using the EfficientNet implementations available in the [timm](https://github.com/rwightman/pytorch-image-models) package and I tried using it with PyTorch XLA. Originally, there was an error, which was resolved in #2652. However, I have now noticed that the training takes extremely slow. For example, using the CIFAR10 notebook, running SE-ResNext50 training only takes about ~35 sec per epoch, while an EfficientNet-B0 takes ~10 minutes epoch! 

Here is a [kernel](https://www.kaggle.com/tanlikesmath/efficientnet-pytorch-xla-bug) replicating the issue, and it should be easy to switch out the timm models from `efficientnet_b0` to `seresnext50_32x4d`.

**Comment by taylanbil** [comment]
do you have an xla metrics report from the run?

**Comment by tmabraham** [comment]
@taylanbil no, but I have just committed a version (version 4) of a 3-epoch run with debug=True and it should be available in 20-30 min at the same link.

**Comment by tmabraham** [comment]
@taylanbil the new kernel version has the output of metrics report:
https://www.kaggle.com/tanlikesmath/efficientnet-pytorch-xla-bug

**Comment by taylanbil** [comment]
Took a quick look. The code suffers from issues discussed in the [troubleshooting doc](https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md)
- `aten::silu_out` is an op that needs to be lowered.
- `aten::_local_scalar_dense` count is growing too aggressively against the `MarkStep` count.

There may be other issues too but these need to be ironed out before proceeding.

**Comment by tmabraham** [comment]
@taylanbil should I open a separate issue for `silu_out` op lowering? 

`aten::_local_scalar_dense count is growing too aggressively against the MarkStep count.` What does this mean regarding the code? Is there something I need to do to prevent this?

**Comment by taylanbil** [comment]
> should I open a separate issue for silu_out op lowering?

yes, that would be great, thanks. Alternatively, here's the [op lowering guide](https://github.com/pytorch/xla/blob/master/OP_LOWERING_GUIDE.md)

The fact that local_scalar_dense is growing too much means the scalar tensor's value it holds is being used in the python code. Either you are logging loss values too often, or you have something like `if tensorvalue > 5`, or `if tensor.any()`, something like that.

**Comment by tmabraham** [comment]
That is interesting that there is something additional regarding scalar tensor use in the python code. I am just using [this code](https://github.com/pytorch/xla/blob/master/contrib/colab/resnet18-training.ipynb), just replacing the ResNet18 model with the timm model. So maybe there is something going on in the timm model.

I have opened an issue for `silu_out` op lowering as shown above.

**Comment by taylanbil** [comment]
Yeah it's possible to have model's forward pass producing `aten::_local_scalar_dense`. I'm not familiar w/ that particular implementation. These should be easily detectable running on one core and putting breakpoints in code, looking at the metrics report at various parts of the forward pass.

**Comment by tmabraham** [comment]
@taylanbil The silu_out op lowering was merged (#2721). Do you just need me to give a metrics report on single core for the timm model forward pass?

**Comment by taylanbil** [comment]
yes, if the slowness persists, that would be great. Make sure to get the latest `torch_xla` so `aten::silu_out`'s are gone.

**Comment by JackCaoG** [comment]
@tmabraham The change merged today morning, you will need to wait for tomorrow's nightly.

**Comment by tmabraham** [comment]
I tested it now, and it looks like the bug is resolved:
https://www.kaggle.com/tanlikesmath/efficientnet-pytorch-xla-bug-resolved

I have also tested an EfficientNet-B3 on a Kaggle competition dataset and it runs with no error at acceptable speeds. Thanks @taylanbil @JackCaoG @tyoc213!

**Comment by taylanbil** [comment]
are there still lots of aten::_local_scalar_dense 's in the metrics report? if you address the growth of sthise you will get a better performance.
